ops/_private/timeconv.py

```diff
@@ -50,6 +51,10 @@ def parse_rfc3339(s: str) -> datetime.datetime:
         tz = datetime.timezone(tz_delta if sign == '+' else -tz_delta)
 
     microsecond = round(float(sfrac or '0') * 1000000)
+    carry, microsecond = divmod(microsecond, 1000000)
```

Comment:
This is nice. However, given that Python 3.11+ truncates, and we'll want to start using that (and this whole function will go away then), I think we should do what it does. It'll also be harder to get wrong with just `int()`. Will this do it?

```python
microsecond = int(float(sfrac or '0') * 1000000)
```